.github/workflows/cypress-update-snapshots.yml

```diff
@@ -0,0 +1,79 @@
+name: Cypress update snapshots
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      testname:
+        description: "Enter the test name to run (see e2e/specs/[testname].spec.js). For example: st_alert"
+        default: "st_alert"
+        type: "string"
+
+# Avoid duplicate workflows on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-cypress-update-snapshots
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash --login -eo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
```

Comment:
Since the `ref` isn't defined in the inputs context above, I think this should be `ref: ${{ github.ref_name }}`? [Example](https://github.com/streamlit/streamlit/blob/develop/.github/workflows/release-candidate.yml#L43) from release candidate